### PR TITLE
Added maxInstances & minInstances to cloudfunction config

### PR DESCRIPTION
**Pull request by AndrewBarber**
Closes issue #55

### CODE REVIEW

File: stacks/gcp/GcpStack.ts [hunk 1]
```diff
@@ -83,6 +83,8 @@ export default class GcpStack extends TerraformStack {
       sourceArchiveObject: object.name,
       availableMemoryMb: func.memory ?? 256,
       entryPoint: "default",
+      maxInstances: func.maxInstances ?? 0,
```
Comment (S48-Mo) [inline stacks/gcp/GcpStack.ts:86]: Why would the min have a default of 1 yet the max have a default of 1?

**Review by S48-Mo — CHANGES_REQUESTED** [review]
Why would the min have a default of 1 yet the max have a default of 0?

**Comment by AndrewBarber** [comment]
> 



> Why would the min have a default of 1 yet the max have a default of 0?

Oops! I misread the line "enter a number greater than or equal to 1" on the [documentation](https://cloud.google.com/functions/docs/configuring/min-instances). Using 0 will clear and set to default, so you are correct.

Updated PR.

File: types/runtime.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ export type FunctionConfig = {
   memory?: number;
   retryOnFailure?: boolean;
   staticIp?: boolean;
+  maxInstances?: number | undefined;
```
Comment (seanpertet) [inline types/runtime.ts:15]: no need to add `| undefined` at the end. The question mark after the key means that the property could be undefined

**Review by seanpertet — APPROVED** [review]

**Review by AndrewBarber — COMMENTED** [review]
Updated maxInstances && minInstances

**Review by S48-Mo — APPROVED** [review]

**Pull request merged**